src/editor/navigation.ts

```diff
@@ -0,0 +1,190 @@
+
+        const step = diagnostic?.path
+            .filter(elem => elem.kind === AnalysisPathKind.Event)[stepIndex] as AnalysisPathEvent;
+        const stepLocation = step?.location ?? diagnosticLocation;
+
+        const targetFile = stepLocation !== undefined ? Uri.file(diagnostic.files[stepLocation.file])
+            : diagnosticLocation !== undefined ? Uri.file(diagnostic.files[diagnosticLocation.file])
+                : file;
+
+        window.showTextDocument(targetFile, {
+            selection: stepLocation !== undefined ? new Range(
+                stepLocation.line - 1,
+                stepLocation.col - 1,
+                stepLocation.line - 1,
+                stepLocation.col - 1
+            ) : undefined
+        });
+
+        if (diagnostic === undefined) {
+            window.showInformationMessage('Unable to find specified report, opened its file instead');
+        } else if (step === undefined) {
+            window.showInformationMessage('Unable to find specified reproduction step, opened the report instead');
+        }
+    }
+
+    getStepIndexUnderCursor(which: 'first' | 'last'): number | null {
+        if (window.activeTextEditor === undefined || ExtensionApi.diagnostics.selectedEntry === undefined) {
+            return null;
+        }
+
+        const cursor = window.activeTextEditor.selection.anchor;
+
+        const entry = ExtensionApi.diagnostics.selectedEntry.diagnostic;
+        const reprPath = entry.path
+            .filter(e => e.kind === AnalysisPathKind.Event) as AnalysisPathEvent[];
+
+        let foundIdx = null;
+
+        for (const [idx, path] of reprPath.entries()) {
+            // Check location first
+            if (window.activeTextEditor.document.uri.fsPath !== entry.files[path.location.file]) {
+                continue;
+            }
+
+            if (cursor.isEqual(new Position(path.location.line-1, path.location.col-1))) {
+                foundIdx = idx;
+
+                if (which === 'first') {
+                    break;
+                }
+
+                continue;
+            }
+
+            // Check inside the ranges
+            if (!path.ranges) {
+                continue;
+            }
+
+            for (const [start, end] of path.ranges) {
+                const range = new Range(
+                    start.line-1,
+                    start.col-1,
+                    end.line-1,
+                    end.col,
+                );
+
+                if (range.contains(cursor)) {
+                    foundIdx = idx;
+
+                    if (which === 'first') {
+                        break;
+                    }
+                }
+            }
+        }
+
+        return foundIdx;
+    }
+
+    nextStep() {
+        const stepIdx = this.getStepIndexUnderCursor('last');
+        const entry = ExtensionApi.diagnostics.selectedEntry?.diagnostic;
+
+        if (stepIdx === null || !entry) {
+            return;
```

Comment:
- If I open a file and the Codechecker view but I do not select any report and I hit ctrl+f7 this function will be called but entry will be null and we just simply return from this function but I didn't get any message /popup that says I have to select a report to jump to the next step.
- Also if I open a bug / clicking to the `Jump to report` button or to a bug step (selecting a report / bug step) and using ctrl+f7, it will do the same thing, so nothing.
- If I select a report and clicking to the `Show reproduction steps` button and then using the ctrl+f7 key combination the feature works.

Reply:
This is intentional, much like how you cannot `Jump to next error in file` if there are no errors in the file... :D
The 'first/last reproduction step' display is also hidden if the path is not shown.

I can set a condition for the hotkey, but this early return leads to a much simpler logic.

Reply:
If I don't select a bug step, it is not annoying if this feature do nothing. But in the second case if I select the first bug step and the report has multiple bug step I expect that this feature should work.

Reply:
Fixed so the repr. path is shown when the user jumps to a step.
It is very unintuitive, though, because I don't have a simple logic for automatically removing a shown path.
It works for now, but the logic will need a rework as part of #10.